### PR TITLE
Enable mining in CLI

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

This allows the user to enable and disable mining in the CLI via a command

## Motivation and Context

We want the user to be able to enable mining in the UI without the user needing to restart the node and set it in the config

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: applications/tari_base_node/src/parser.rs [hunk 1]
```diff
@@ -210,6 +216,9 @@ impl Parser {
             ListConnections => {
                 self.process_list_connections();
             },
+            EnableMining => {
+                self.process_enable_mining();
```
Comment (CjS77) [inline applications/tari_base_node/src/parser.rs]:
```suggestion
                self.process_toggle_mining();
```

File: applications/tari_base_node/src/parser.rs [hunk 2]
```diff
@@ -303,6 +312,12 @@ impl Parser {
         });
     }
 
+    fn process_enable_mining(&mut self) {
```
Comment (CjS77) [inline applications/tari_base_node/src/parser.rs]:
```suggestion
    fn process_toggle_mining(&mut self) {
```

File: applications/tari_base_node/src/parser.rs [hunk 3]
```diff
@@ -174,6 +177,9 @@ impl Parser {
             ListConnections => {
                 println!("Lists the peer connections currently held by this node");
             },
+            EnableMining => {
+                println!("Enable or disable the miner on this node, calling this command will swop the state");
```
Comment (CjS77) [inline applications/tari_base_node/src/parser.rs]:
```suggestion
                println!("Enable or disable the miner on this node, calling this command will toggle the state");
```

File: applications/tari_base_node/src/parser.rs [hunk 4]
```diff
@@ -69,6 +69,7 @@ pub enum BaseNodeCommand {
     ListPeers,
     ListConnections,
     Whoami,
+    EnableMining,
```
Comment (CjS77) [inline applications/tari_base_node/src/parser.rs]:
```suggestion
    ToggleMining,
```

File: applications/tari_base_node/src/parser.rs [hunk 5]
```diff
@@ -179,6 +182,9 @@ impl Parser {
             ListHeaders => {
                 println!("List the last headers up to a maximum of 10 of the current chain");
             },
+            EnableMining => {
```
Comment (CjS77) [inline applications/tari_base_node/src/parser.rs]: How is this compiling?

Comment (CjS77) [reply applications/tari_base_node/src/parser.rs]: Intersting.
```
 EnableMining  => {
    |------------ matches any value
```
which is very misleading, but basically, this match statement is very broken.

Comment (SWvheerden) [reply applications/tari_base_node/src/parser.rs]: mmm interesting, I specifically tried to run a full tests at this to capture stuff like this...

Comment (SWvheerden) [reply applications/tari_base_node/src/parser.rs]: updated

**Pull request merged**